### PR TITLE
Completed Trees-3 || Please Review

### PR DESCRIPTION
**Pull request by soumil100**

### CODE REVIEW

File: Problem1.java [hunk 1]
```diff
@@ -0,0 +1,76 @@
+//Time Complexity :O(n*n) = O(n^2)
```
Comment (AbhijeetDhillon) [inline Problem1.java:1]: Could you explain the time complexity? According to me, it should be O(n)

**Comment by AbhijeetDhillon** [comment]
Nice work Soumil, check the inline comment and let me know what do you think about it.

**Comment by AbhijeetDhillon** [comment]
Nice work Soumil, check the inline comment and let me know what do you think about it.

File: Problem1.java (hunk 1 above)
Comment (soumil100) [reply Problem1.java:1]: For each element of the node, you are creating a copy of the paths which takes O(n) worst-case. That's why n* n. A better answer would be n * h where h is the height of the tree. For every node, we need a copy array of h elements. So Time Complexity will be O(n*h).

**Comment by soumil100** [comment]
Checked on Leet code Premium it states, Time Complexity is O(n^2)

**Pull request closed without merging**